src/utils/cloneObject.test.ts

```diff
@@ -0,0 +1,34 @@
+import cloneObject from './cloneObject';
+
+describe('clone', () => {
+  it('should clone object and not mutate the orginal object', () => {
+    const data = {
+      test: {
+        test: '12',
+        test1: [1, 2, 3, 4],
+        date: new Date('2020-10-15'),
+        deep: {
+          date: new Date('2020-10-15'),
+        },
+      },
+    };
+
+    const copy = cloneObject(data);
+    expect(cloneObject(data)).toEqual(copy);
+
+    // @ts-ignore
+    copy.test.what = '1243';
+    copy.test.date = new Date('2020-10-116');
+
+    expect(data).toEqual({
+      test: {
+        test: '12',
+        test1: [1, 2, 3, 4],
+        date: new Date('2020-10-15'),
+        deep: {
+          date: new Date('2020-10-15'),
+        },
+      },
+    });
```

Comment:
Just adding to the verboseness of the test. May not be perfectly formatted.
```suggestion
    const data = {
      test: {
        date: new Date('2020-10-15'),
        test0: 12,
        test1: '12',
        test2: [1, 2, 3, 4],
        deep: {
          date: new Date('2020-10-15'),
          test0: 12,
          test1: '12',
          test2: [1, 2, 3, 4],
        },
      },
    };

    const copy = cloneObject(data);
    expect(cloneObject(data)).toEqual(copy);

    // @ts-ignore
    copy.test.what = '1243';
    copy.test.date = new Date('2020-10-116');

    expect(data).toEqual({
      test: {
        date: new Date('2020-10-15'),
        test0: 12,
        test1: '12',
        test2: [1, 2, 3, 4],
        deep: {
          date: new Date('2020-10-15'),
          test0: 12,
          test1: '12',
          test2: [1, 2, 3, 4],
        },
      },
    });
```